src/System.Net.Http/src/System/Net/Http/Unix/CurlCallbacks.cs

```diff
@@ -0,0 +1,320 @@
+                    // headers get merged into a single line
+                    string responseHeader = Marshal.PtrToStringAnsi(ptr).Trim();
+                    HttpResponseMessage response = state.ResponseMessage;
+
+                    // TODO: Understand scenarios where multiple sets of headers are received
+                    if (!TryParseStatusLine(response, responseHeader))
+                    {
+                        int colonIndex = responseHeader.IndexOf(':');
+
+                        // Skip malformed header lines that are missing the colon character.
+                        if (colonIndex > 0)
+                        {
+                            string headerName = responseHeader.Substring(0, colonIndex);
+                            string headerValue = responseHeader.Substring(colonIndex + 1);
+
+                            if (!response.Headers.TryAddWithoutValidation(headerName, headerValue))
+                            {
+                                // TODO: Skip compression related headers
+                                response.Content.Headers.TryAddWithoutValidation(headerName, headerValue);
+                            }
+                        }
+                    }
+                    return totalSize;
+                });
+        }
+
+        private unsafe static ulong CurlReceiveBodyCallback(
+            byte* buffer,
+            ulong size,
+            ulong nitems,
+            IntPtr context)
+        {
+            size *= nitems;
+            GCHandle gch = GCHandle.FromIntPtr(context);
+            var state = (RequestCompletionSource)gch.Target;
+
+            try
+            {
+                // Set task completion after all headers have been received
+                if (!state.Task.IsCompleted && !state.Task.IsCanceled && !state.Task.IsFaulted)
+                {
+                    state.TrySetResult(state.ResponseMessage);
+                }
+
+                // TODO: Handle compression and chunked-mode
+
+                // Wait for a reader
+                CurlResponseStream contentStream = state.ResponseMessage.ContentStream;
+                contentStream.WaitAndSignalReaders(buffer, (long)size);
+                return size;
+            }
+            catch (Exception ex)
+            {
+                HandleAsyncException(state, ex);
+                return (size > 0) ? size - 1 : 1; // Force curl_easy_perform to fail
+            }
+        }
+
+        private static ulong CurlSendCallback(
+            IntPtr buffer,
+            ulong size,
+            ulong nitems,
+            IntPtr context)
+        {
+            return ExecuteCallback(buffer, size, nitems, context,
+                (ptr, totalSize, state) =>
+                {
+                    state.CancellationToken.ThrowIfCancellationRequested();
+                    Stream contentStream = state.RequestContentStream;
+
+                    // TODO: Handle chunked-mode
```

Comment:
In the meantime, is there a quick check we could do to throw a NotImplementedException if such missing functionality is hit?